src/client/browserType.ts

```diff
@@ -109,12 +109,14 @@ export class BrowserType extends ChannelOwner<channels.BrowserTypeChannel, chann
 
   async connect(params: ConnectOptions): Promise<Browser> {
     const logger = params.logger;
+    const paramsHeaders = params.headers ? params.headers : {};
+    paramsHeaders['User-Agent'] = getUserAgent();
```

Comment:
This is way it cannot be overridden by the client, let's change it to something like `Object.assign({'User-Agent': getUserAgent()}, params.headers)`

Reply:
Resolved. Committed the change